lib/CourseThemeModule.js

```diff
@@ -132,64 +132,50 @@ class CourseThemeModule extends AbstractApiModule {
+    // font files
+    if (fontData._items) {
       await Promise.all(fontData._items.map(async f => {
         if (!f['font-family']) return
-
         // copy each uploaded font file
         for (const font in f._files) {
           const [data] = await assets.find({ _id: f._files[font] })
           const asset = assets.createFsWrapper(data)
           try {
-            await fs.writeFile(`${fontDest}/${f._files[font]}.${asset.data.subtype}`, await asset.read())
-
+            const relativeFontPath = `fonts/${f._files[font]}.${asset.data.subtype}`
+            const absoluteFontPath = `${fwBuild.dir}/src/theme/${fwBuild.courseData.config.data._theme}/${relativeFontPath}`
+            await fs.writeFile(absoluteFontPath, await asset.read())
             // construct font import styling
-            fontImportCSS += `@font-face {\n\tfont-family: ${f['font-family']};\n\tsrc: url('./fonts/${f._files[font]}.${asset.data.subtype}') format('${asset.data.subtype}');\n\t${fontWeightCheck[font]()}\n}\n`
+            fontImportCSS += `@font-face ${JSON.stringify({
+              'font-family': `${f['font-family']};`,
+              src: `url('./${relativeFontPath}') format('${asset.data.subtype}');`,
+              'font-weight': `${font.includes('light') ? 'light' : font.includes('bold') ? 'bold' : 'normal'};`,
+              'font-style': `${font.includes('Italic') ? 'italic' : 'normal'};`
+            }, null, 2)}`
           } catch (e) {
             this.log('error', `failed to write ${f._files[font]}, ${e.message}`)
           }
         }
       }))
-
-      fwBuild.courseData.course.data.themeVariables._font._items = undefined
+      delete fontData._items
     }
-    // Instruction style checkbox
-    if (typeof fwBuild.courseData.course.data.themeVariables._font._fontAssignments._instruction['instruction-font-style'] !== null) {
-      fwBuild.courseData.course.data.themeVariables._font._fontAssignments._instruction['instruction-font-style'] = fwBuild.courseData.course.data.themeVariables._font._fontAssignments._instruction._isInstructionItalic ? 'italic' : 'normal'
-
-      fwBuild.courseData.course.data.themeVariables._font._fontAssignments._instruction._isInstructionItalic = undefined
+    // instruction style checkbox
+    const _instruction = fontData._fontAssignments._instruction
+    if (_instruction['instruction-font-style']) {
+      _instruction['instruction-font-style'] = _instruction._isInstructionItalic ? 'italic' : 'normal'
+      delete _instruction._isInstructionItalic
     }
-    // External fonts
-    if (fontData._externalFonts !== null) {
-      await Promise.all(fontData._externalFonts.map(async f => {
+    // external fonts
+    if (fontData._externalFonts) {
+      fontData._externalFonts.forEach(f => {
         if (f) fontImportCSS += `@import url('${f}');\n`
-      }))
-      fwBuild.courseData.course.data.themeVariables._font._externalFonts = undefined
+      })
+      delete fontData._externalFonts
```

Comment:
In the above changes, I do the following:
- Store a local reference to fontData's _instruction attribute to allow more readable referencing
- Replace code setting attributes to undefined with a `delete`
- Remove the Promise.all, as the code doesn't require it